lib/praxis/api_general_info.rb

```diff
@@ -40,14 +48,32 @@ def description(val=nil)
       end
     end
 
+    def version_with(val=nil)
+      if val.nil?
+        get(:version_with)
+      else
+        if @global_info.nil? # this *is* the global info
+          Application.instance.versioning_scheme = val
+          set(:version_with, val)
+        else
+          raise 'Can only use `version_with` in global api info.'
```

Comment:
I'd change it to be more descriptive:
"Use of `version_with` is only allowed in the global part of the API definition (but you are attempting to use it in the API definition of version #{self.version}"